app/process_proposal.go

```diff
@@ -122,24 +125,40 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 
 	}
 
-	// Construct the data square from the block's transactions
-	dataSquare, err := square.Construct(
-		req.BlockData.Txs,
-		app.MaxEffectiveSquareSize(sdkCtx),
-		subtreeRootThreshold,
+	var (
+		dataSquareBytes [][]byte
+		err             error
 	)
-	if err != nil {
-		logInvalidPropBlockError(app.Logger(), req.Header, "failure to compute data square from transactions:", err)
+
+	switch app.AppVersion() {
+	case v3:
+		var dataSquare squarev2.Square
```

Comment:
should we add a unit test to process proposal to verify this logic?

Reply:
What logic would this test be verifying? Do we want to build a v3 square with an authored blob and assert that we can parse out the signer or something like that?